tmt/steps/report/reportportal.py

```diff
@@ -143,8 +153,82 @@ def go(self) -> None:
             message = response.text
         if not response.ok:
             raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
+                "Received non-ok status code from ReportPortal, "
+                f"response text is: {message}")
         else:
             self.debug(f"Response code from the server: {response.status_code}")
             self.debug(f"Message from the server: {message}")
             self.info("report", "Successfully uploaded.", "yellow")
+
+        try:
+            _ = json.dumps(meta_data)
+        except Exception as exc:
+            raise tmt.utils.SpecificationError("Invalid syntax of provided metadata") from exc
+
+        if meta_data:
+            uuid = None
+            re_result = re.search("\\w{8}-(\\w{4}-){3}\\w{12}", str(message))
+            if isinstance(re_result, re.Match):
+                uuid = re_result.group()
+
+            if not isinstance(uuid, str):
+                self.debug(f"Could not update metadata in {message}")
+                return
```

Comment:
This is a much less common flow: `re_result` can be either `Match` or `None`, therefore `isinstance` is not needed and the default boolean check would do, plus if there was a match, the group 0 would exist, therefore checking `uuid` for string-ness is also not needed, and an early return would take care of the negative branches. How about this version?

```python
match = re.search(r"\w{8}-(\w{4}-){3}\w{12}", str(message))

if not match:
    self.debug(f"Could not update metadata in {message}")
    return

uuid = match.group()
```

Also note the use of raw string in my proposal, `r'...'` - simplifies the backslash situation in the string, it's usually how regular expressions should be written in Python lands.

Reply:
Ack